flags.go

```diff
@@ -17,28 +20,71 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// A NetEnumerator enumerates local IP addresses.
+type NetEnumerator struct {
+	Interfaces     func() ([]net.Interface, error)
+	InterfaceAddrs func(*net.Interface) ([]net.Addr, error)
+}
+
+type IPConfiguration struct {
+	IpAddress    string `json:"ipaddress"`
+	Netmask      string `json:"netmask"`
+	Gateway      string `json:"gateway"`
+	PrimaryDns   string `json:"primarydns"`
+	SecondaryDns string `json:"secondarydns"`
+}
+
+func makeIpVar(val *string) *IpVar {
+	return &IpVar{val}
+}
+
+type IpVar struct {
+	val *string
+}
+
+func (a *IpVar) String() string {
+	if a.val == nil {
+		return ""
+	}
+	return *a.val
+}
+func (a *IpVar) Set(val string) error {
+	if net.ParseIP(val) == nil {
+		return errors.New("not a valid ip address")
+	}
+	*a.val = val
+	return nil
+}
```

Comment:
why not just a single function that checks if the IP is valid? This seems a bit heavy?

Reply:
Wanted validation to be handled by the library with the variable registration and that was the best example implementation I could find at the time and my library/golang expertise could grok. Another search with more experience, found a better approach.